### PR TITLE
CI-70: Content of the Access Site page is shown incorrect in Gmail/Outlook (not MailTrap)

### PR DESCRIPTION
**Pull request by VitaliiShevchenko**
[CI-70](https://cloverpop-internship.atlassian.net/browse/CI-70)

[CI-70]: https://cloverpop-internship.atlassian.net/browse/CI-70?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SokalMet — COMMENTED** [review]
as I see - this PR looks the same as here https://github.com/wahanegi/vibereport/pull/23#pullrequestreview-1358787464. I think you do not need to make copies of the same changes. Once prev PR will be done, merge it from master here, and will see then

**Comment by VitaliiShevchenko** [comment]
Ok. I will be change in the master, and after that make a necessary PR. Thanks

File: app/mailers/user_email_mailer.rb [hunk 1]
```diff
@@ -7,11 +7,11 @@ class UserEmailMailer < ApplicationMailer
...
     @user = user
-    @general_link = { controller: 'api/v1/responses', action: 'response_flow_from_email',
-                      time_period_id: TimePeriod.current, not_working: false, user_id: @user.id }
-    @link_for_own_word = [@general_link, { last_step: 'emotion-entry' }].reduce(:merge!)
-    @link_for_was_not  = [@general_link, { last_step: 'results' }].reduce(:merge!)
-    @link_for_emotion  = [@general_link, { emotion_id: nil, last_step: 'meme-selection' }].reduce(:merge!)
+    general_link = { controller: 'api/v1/responses', action: 'response_flow_from_email',
+                     time_period_id: TimePeriod.current, not_working: false, user_id: user.id }
+    @link_for_own_word = [general_link, { last_step: 'emotion-entry' }].reduce(:merge)
+    @link_for_was_not  = [general_link, { last_step: 'results' }].reduce(:merge)
```
Comment (SBorozenets) [inline app/mailers/user_email_mailer.rb:13]: the same

**Review by SBorozenets — CHANGES_REQUESTED** [review]

**Comment by SBorozenets** [comment]
@VitaliiShevchenko it's not the best practice using the magic number in methods, I think we can give up using the method `def calculation(index)` and rewrite code from
```
@table = []
(0..@emotions.length - 1).each { |index| @table.push(@emotions[calculation(index)]) }


def calculation(index)
    (index - (6 * (index / 6).ceil - 1)) * 6 - ((index / 6).ceil - 1) - 2
  end
```
 to smth like this
```
@table = []
@emotions.each_slice(6) { |sliced_emotions| @table << sliced_emotions }
@table = @table.transpose.flatten
```

**Review by SokalMet — APPROVED** [review]

File: app/mailers/user_email_mailer.rb [hunk 2]
```diff
@@ -7,11 +7,11 @@ class UserEmailMailer < ApplicationMailer
...
+    @link_for_emotion  = [general_link, { emotion_id: nil, last_step: 'meme-selection' }].reduce(:merge)
     @view_complete_by = range_format(time_period)
     @emotions = Emotion.positive.sample(NUMBER_OF_ELEMENTS) +
```
[the remark this answers is not in the document]
Comment (VitaliiShevchenko) [reply app/mailers/user_email_mailer.rb:16]: Excuse me I don't understood what you mean. If you mean variables `@link_for_own_word,  @link_for_was_not and @link_for_emotion ` that variables using by creating email letters in the method `response_invite.html.erb`

File: app/mailers/user_email_mailer.rb [hunk 3]
```diff
@@ -7,11 +7,11 @@ class UserEmailMailer < ApplicationMailer
 
   def response_invite(user, time_period)
     @user = user
```
[the remark this answers is not in the document]
Comment (VitaliiShevchenko) [reply app/mailers/user_email_mailer.rb:9]: `@user` this variable using to create email letters in the method `response_invite.html.erb`

File: app/mailers/user_email_mailer.rb [hunk 4]
```diff
@@ -7,11 +7,11 @@ class UserEmailMailer < ApplicationMailer
...
-    @general_link = { controller: 'api/v1/responses', action: 'response_flow_from_email',
-                      time_period_id: TimePeriod.current, not_working: false, user_id: @user.id }
-    @link_for_own_word = [@general_link, { last_step: 'emotion-entry' }].reduce(:merge!)
-    @link_for_was_not  = [@general_link, { last_step: 'results' }].reduce(:merge!)
-    @link_for_emotion  = [@general_link, { emotion_id: nil, last_step: 'meme-selection' }].reduce(:merge!)
+    general_link = { controller: 'api/v1/responses', action: 'response_flow_from_email',
+                     time_period_id: TimePeriod.current, not_working: false, user_id: user.id }
+    @link_for_own_word = [general_link, { last_step: 'emotion-entry' }].reduce(:merge)
```
[the remark this answers is not in the document]
Comment (VitaliiShevchenko) [reply app/mailers/user_email_mailer.rb:12]: Thank you very much for this. It looks very nice and better than my)

File: app/mailers/user_email_mailer.rb (hunk 1 above)
Comment (VitaliiShevchenko) [reply app/mailers/user_email_mailer.rb]: Thanks

File: app/mailers/user_email_mailer.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (SBorozenets) [reply app/mailers/user_email_mailer.rb:16]: @VitaliiShevchenko sorry for misunderstood I wrote about @emotions

File: app/mailers/user_email_mailer.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (SBorozenets) [reply app/mailers/user_email_mailer.rb:9]: I don't see any usage of `@user` in `response_invite.html.erb`, I see that you use it just on row 22 of this file, you can write 
`mail(to: user.email, subject: "Hey #{user.first_name}, how was your week?")`

**Review by rogergraves — COMMENTED** [review]
I agree with Svitlana, please only declare instance variables if they will be used. Otherwise they take up memory unnecessarily and go against the best practices of OOP.

File: app/mailers/user_email_mailer.rb [hunk 5]
```diff
@@ -8,25 +8,22 @@ class UserEmailMailer < ApplicationMailer
   def response_invite(user, time_period)
     @user = user
     @general_link = { controller: 'api/v1/responses', action: 'response_flow_from_email',
-                      time_period_id: TimePeriod.current, not_working: false, user_id: @user.id }
-    @link_for_own_word = [@general_link, { last_step: 'emotion-entry' }].reduce(:merge!)
-    @link_for_was_not  = [@general_link, { last_step: 'results' }].reduce(:merge!)
-    @link_for_emotion  = [@general_link, { emotion_id: nil, last_step: 'meme-selection' }].reduce(:merge!)
+                     time_period_id: TimePeriod.current, not_working: false, user_id: user.id }
```
Comment (rogergraves) [inline app/mailers/user_email_mailer.rb:11]: CodeClimate says to align this with the keys of a hash literal

File: app/mailers/user_email_mailer.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (VitaliiShevchenko) [reply app/mailers/user_email_mailer.rb:16]: Ok, thanks for detail explanation)

File: app/mailers/user_email_mailer.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (VitaliiShevchenko) [reply app/mailers/user_email_mailer.rb]: I agree with you. Excuse me I mistaked

File: app/mailers/user_email_mailer.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (VitaliiShevchenko) [reply app/mailers/user_email_mailer.rb]: only must was without @ )))

**Review by rogergraves — APPROVED** [review]
👍🏻

**Review by SBorozenets — APPROVED** [review]

**Review by SerhiiBorozenets — APPROVED** [review]

**Pull request merged**